alert-engine/src/main/scala/io/phdata/pulse/alertengine/DefaultAlertEngine.scala

```diff
@@ -63,20 +67,23 @@ class DefaultAlertEngine(solrTrigger: SolrAlertTrigger,
    */
   def triggeredAlert(applicationName: String, alertRule: AlertRule): Option[TriggeredAlert] =
     alertRule.alertType match {
-      case Some(AlertTypes.solr) => solrTrigger.check(applicationName, alertRule)
-      case Some(AlertTypes.sql) =>
-        if (sqlTrigger.isDefined) {
-          sqlTrigger.get.check(applicationName, alertRule)
-        } else {
-          logger.warn(s"No sql alert trigger available to process the alert rule: $alertRule")
-          None
-        }
-      case None => solrTrigger.check(applicationName, alertRule)
+      case Some(AlertTypes.solr) => triggerOrElse(solrTrigger, applicationName, alertRule)
+      case Some(AlertTypes.sql) => triggerOrElse(sqlTrigger, applicationName, alertRule)
+      case None => triggerOrElse(solrTrigger, applicationName, alertRule)
       case _ =>
         logger.warn(s"Unknown alert type in alert rule: $alertRule")
         None
     }
 
+  private def triggerOrElse(trigger: Option[AlertTrigger],
+                            applicationName: String,
+                            alertRule: AlertRule): Option[TriggeredAlert] =
+    trigger.map(_.check(applicationName, alertRule)).getOrElse {
+      logger.warn(
```

Comment:
`AlertEngineMain` ensures that when the alert engine is created, it will try to create all requested alert types. If one fails to create you will get an exception.

Would you rather log an error or throw an exception? It should only be possible if somebody were to use the `DefaultAlertEngine` class directly.